elasticsearch/src/main/java/org/hibernate/search/elasticsearch/analyzer/impl/ScopedElasticsearchAnalyzerReference.java

```diff
@@ -6,26 +6,161 @@
+			throw LOG.lazyRemoteAnalyzerReferenceNotInitialized( this );
+		}
+		return analyzer;
+	}
+
+	public boolean isInitialized() {
+		return analyzer != null;
+	}
+
+	public void initialize() {
+		if ( this.analyzer != null ) {
+			throw new AssertionFailure( "An Elasticsearch analyzer reference has been initialized more than once:" + this );
+		}
+		this.analyzer = builder.buildAnalyzer();
+		this.builder = null;
 	}
 
 	@Override
-	public <T extends AnalyzerReference> boolean is(Class<T> analyzerType) {
-		return analyzerType.isAssignableFrom( ScopedElasticsearchAnalyzerReference.class );
+	public void close() {
+		if ( isInitialized() ) {
+			getAnalyzer().close();
+		}
+	}
+
+	@Override
+	public CopyBuilder startCopy() {
+		return new CopyBuilder( getAnalyzer() );
+	}
+
+	/**
+	 * A builder that defers the actual analyzer creation to later during the search
+	 * factory initialization, so that the builder accepts dangling references.
+	 *
+	 * @author Yoann Rodiere
+	 */
+	public static class DeferredInitializationBuilder implements ScopedAnalyzerReference.Builder {
+
+		private ElasticsearchAnalyzerReference globalAnalyzerReference;
+		private final Map<String, ElasticsearchAnalyzerReference> scopedAnalyzerReferences = new HashMap<>();
+
+		public DeferredInitializationBuilder(ElasticsearchAnalyzerReference globalAnalyzerReference,
+				Map<String, ElasticsearchAnalyzerReference> scopedAnalyzers) {
+			this.globalAnalyzerReference = globalAnalyzerReference;
+			this.scopedAnalyzerReferences.putAll( scopedAnalyzers );
+		}
+
+		@Override
+		public ElasticsearchAnalyzerReference getGlobalAnalyzerReference() {
+			return globalAnalyzerReference;
+		}
+
+		@Override
+		public void setGlobalAnalyzerReference(AnalyzerReference globalAnalyzerReference) {
+			this.globalAnalyzerReference = getElasticsearchAnalyzerReference( globalAnalyzerReference );
+		}
+
+		@Override
+		public void addAnalyzerReference(String scope, AnalyzerReference analyzerReference) {
+			scopedAnalyzerReferences.put( scope, getElasticsearchAnalyzerReference( analyzerReference ) );
+		}
+
+		@Override
+		public ScopedElasticsearchAnalyzerReference build() {
+			return new ScopedElasticsearchAnalyzerReference( this );
+		}
+
+		protected final ScopedElasticsearchAnalyzer buildAnalyzer() {
+			ElasticsearchAnalyzer globalAnalyzer = globalAnalyzerReference.getAnalyzer();
+
+			Map<String, ElasticsearchAnalyzer> scopedAnalyzers = new HashMap<>();
+
+			for ( Map.Entry<String, ElasticsearchAnalyzerReference> entry : scopedAnalyzerReferences.entrySet() ) {
+				scopedAnalyzers.put( entry.getKey(), entry.getValue().getAnalyzer() );
+			}
+
+			return new ScopedElasticsearchAnalyzer( globalAnalyzer, scopedAnalyzers );
+		}
+	}
+
+	public static class CopyBuilder implements ScopedAnalyzerReference.CopyBuilder {
+
+		private final ScopedElasticsearchAnalyzer baseAnalyzer;
+		private final Map<String, ElasticsearchAnalyzerReference> scopedAnalyzerReferences = new HashMap<>();
```

Comment:
Let's work directly on the analyzers here. We don't need to be on the reference level.